// athom-api/index.mjs
import fetch from './src/lib/util';
import './src/lib/types';
import HomeyAPI from './src/lib/HomeyAPI';
import AthomCloudAPI from './src/lib/AthomCloudAPI';
import AthomAppsAPI from './src/lib/AthomAppsAPI';
import AthomCallbackAPI from './src/lib/AthomCallbackAPI';
import AthomWebhooksAPI from './src/lib/AthomWebhooksAPI';
import AthomNotificationsAPI from './src/lib/AthomNotificationsAPI';
import AthomConnectAPI from './src/lib/AthomConnectAPI';
import AthomFirmwareAPI from './src/lib/AthomFirmwareAPI';
import AthomDNSAPI from './src/lib/AthomDNSAPI';
import AthomSetupAPI from './src/lib/AthomSetupAPI';
import AthomGeolocateAPI from './src/lib/AthomGeolocateAPI';
import AthomBackupAPI from './src/lib/AthomBackupAPI';
import AthomFlowAPI from './src/lib/AthomFlowAPI';
import AthomStoreAPI from './src/lib/AthomStoreAPI';
import AthomWeatherAPI from './src/lib/AthomWeatherAPI';
import AthomStorageAdapter from './src/lib/StorageAdapter';
import EventEmitter from './src/lib/util/EventEmitter';

if (typeof window === 'object') {
  window.HomeyAPI = HomeyAPI;
  window.AthomCloudAPI = AthomCloudAPI;
  window.AthomAppsAPI = AthomAppsAPI;
  window.AthomCallbackAPI = AthomCallbackAPI;
  window.AthomWebhooksAPI = AthomWebhooksAPI;
  window.AthomNotificationsAPI = AthomNotificationsAPI;
  window.AthomConnectAPI = AthomConnectAPI;
  window.AthomFirmwareAPI = AthomFirmwareAPI;
  window.AthomDNSAPI = AthomDNSAPI;
  window.AthomSetupAPI = AthomSetupAPI;
  window.AthomGeolocateAPI = AthomGeolocateAPI;
  window.AthomBackupAPI = AthomBackupAPI;
  window.AthomFlowAPI = AthomFlowAPI;
  window.AthomStoreAPI = AthomStoreAPI;
  window.AthomWeatherAPI = AthomWeatherAPI;
  window.AthomStorageAdapter = AthomStorageAdapter;
}

export {
  HomeyAPI,
  AthomCloudAPI,
  AthomAppsAPI,
  AthomCallbackAPI,
  AthomWebhooksAPI,
  AthomNotificationsAPI,
  AthomConnectAPI,
  AthomFirmwareAPI,
  AthomDNSAPI,
  AthomSetupAPI,
  AthomGeolocateAPI,
  AthomBackupAPI,
  AthomFlowAPI,
  AthomStoreAPI,
  AthomWeatherAPI,
  EventEmitter,
  AthomStorageAdapter,
};

export default {
  HomeyAPI,
  AthomCloudAPI,
  AthomAppsAPI,
  AthomCallbackAPI,
  AthomWebhooksAPI,
  AthomNotificationsAPI,
  AthomConnectAPI,
  AthomFirmwareAPI,
  AthomDNSAPI,
  AthomSetupAPI,
  AthomGeolocateAPI,
  AthomBackupAPI,
  AthomFlowAPI,
  AthomStoreAPI,
  AthomWeatherAPI,
  EventEmitter,
  AthomStorageAdapter,
};
